### PR TITLE
Ui 427 button

### PR DESCRIPTION
**Pull request by sheela-svmx**
PR comments 

- Josh I have left the button type icon-bare, icon- etc as is. It affects the styles. We need to talk before I proceed.If i just have button types without Icon the styles are affected and i need to know how we should handle this.

### CODE REVIEW

File: lib/components/Button/Button.js [hunk 1]
```diff
@@ -1,39 +1,83 @@
...
-    this.onClick = this.onClick.bind(this);
-  }
+const BUTTON_SIZES = ['x-small', 'small', 'medium', 'large'];
+
+const BUTTON_TYPES = [
+  'neutral',
+  'brand',
+  'destructive',
+  'inverse',
+  'icon-bare',
```
Comment (sheela-svmx) [inline lib/components/Button/Button.js]: If i remove the icon- from here, the styles are affected. these are required for the lightning styles. So until then I am leaving this as is.

**Comment by sheela-svmx** [comment]
@joshsylvester @mallik-svmx .
 - I have made the PR from my forked repo. 
- Josh I have addressed like we discussed except for the Button types which have icon-bare , icon-container etc. These affect the styles of the button rendered and I dont know what needs to . be done. I will discuss with you tomorrow on this.

File: lib/components/Button/Button.js (hunk 1 above)
Comment (joshsylvester) [reply lib/components/Button/Button.js]: leave it - those styles appear tied intrinsically to the lightning button styles.

File: lib/components/Button/Button.js [hunk 2]
```diff
@@ -62,80 +103,15 @@ export default class Button extends Component {
         {...props}
         onClick={this.onClick}
       >
-        { icon && iconAlign !== 'right' ? this.renderIcon(iconSize, inverse) : null }
-        { children || label }
-        { icon && iconAlign === 'right' ? this.renderIcon(iconSize, inverse) : null }
-        { iconMore ? this.renderIconMore() : null }
-        { alt ? <span className="slds-assistive-text">{ alt }</span> : null }
-        { loading ? <Spinner /> : null }
+        {children ? React.Children.map(children, this.constructButtonIconProps) : null }
```
Comment (joshsylvester) [inline lib/components/Button/Button.js]: this, and the few lines below should be converted to use a Logical Operator.

```
{children && React.Children.map(children, this.constructButtonIconProps)}
```

**Comment by mallik-svmx** [comment]
@sheela-svmx Here few review comments as we discussed: 
#1 - Comment propTypes used in documentation
#2 - Provide defaultProps to as many props
#3 - Do not need call/apply, this is taken care for you from ES6 arrow fn
#4 - Use Logical && from Josh comment
#5 - Precise examples, including iconMore etc.,

**Comment by sheela-svmx** [comment]
@mallik-svmx updated

**Pull request merged**